Adopting nRF SoC and boards to HWMv2

tejlmand opened the pull request:
Followup on #50305 with porting of Nordic's nRF SoCs and boards.

rerickson1 reviewed (commented):
Laird Connectivity boards look good.

I noticed `rm1xx_dvk` is missing from being moved to v2. It is based on `SOC_NRF51822_QFAC`

tejlmand commented:
compliance and doc build failures are related to `MAINTAINERS.yml` file.

This file will be independently updated when socs and boards have been ported.

Compliance and doc build errors
> get_maintainer.MaintainersError: MAINTAINERS.yml: glob pattern 'boards/arm/bl65*/' in 'files' in area 'Laird Connectivity platforms' does not match any files

nordicjm commented:
Remaining test failures:

- 96b_carbon needs stm32f4xx porting to hwmv2
- bsim needs soc/boards v2 folder location to be set for the final location so bsim files can be updated

kartben commented on boards/v2/nordic_nrf/thingy53/doc/index.rst:
```diff
@@ -0,0 +1,51 @@
+* a secondary Arm Cortex-M33 core, with a reduced feature set, running at
+  a fixed 64 MHz, referred to as the **network core**.
+
+The ``nrf5340dk/nrf5340/cpuapp`` build target provides support for the application
+core on the nRF5340 SoC. The ``nrf5340dk/nrf5340/cpunet`` build target provides
+support for the network core on the nRF5340 SoC.
```
s/nrf5340dk/thingy53/ ?

nordicjm replied:
Fixed

kartben commented on boards/v2/nordic_nrf/thingy52/doc/index.rst:
https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/boards/nrf/battery/src/battery.c#L26-L37 will need updating

nordicjm committed:
boards: common: openocd-nrf5: Add HWMv2 support

Adds support for determining the running from the identifier
which is needed for HWMv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
kconfig: soc: adopt Nordic SoC series to support hw model v2 scheme

This commit adopts hw model v2 to nRF SoC family.

This ensures that when hw model v2 is used for an nRF51, nRF52, nRF53,
and nRF91 SoC series, so that all SoC selections are handled internally
by Kconfig, and no selection / re-configuration is possible by end-user
or through configuration files.

Hw model v2 requires the SoC to be self-contained, that is no
references are allowed out-side the SoC Kconfig tree.

All Zephyr Kconfig specifics are therefore sourced into the Zephyr
Kconfig tree through Kconfig.zephyr and Kconfig.zephyr.defconfig.

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: bl652_dvk: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: update nRF51dk board to board scheme v2.

This commit updates nrf51dk_nrf51422 to use board scheme v2, and
thus becomes board nrf51dk with the SoC nrf51422.

It also uses the new SoC approach, and thereby ensuring that the right
SoC is always used, and cannot accidentially be changed by the user.

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: Update nrf5340dk_nrf5340 to HWMv2 scheme

This commit updates nrf5340dk_nrf5340 to use board scheme v2, and
thereby remove the Kconfig BOARD setting, as this is now passed from
CMake.

The nRF5340dk now support cpuapp and cpunet as cpusets, and the ns as
a build variant.

It also uses the new SoC approach, and thereby ensuring that the right
SoC is always used, and cannot accidentially be changed by the user.

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: update nRF52840dk_nrf52840/nrf52811 board to board scheme v2.

This commit updates nrf52840dk_nrf52840 and nrf52840_nrf52811 to use
board scheme v2, and thereby remove the Kconfig BOARD setting, as this
is now passed from CMake.

It combines both boards in a common nrf52840dk board folder where each
SoC is an entry in the board.yml file.

It also uses the new SoC approach, and thereby ensuring that the right
SoC is always used, and cannot accidentially be changed by the user.

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: bl654_usb: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: bl654_sensor_board: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: nrf52dk: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: bl654_dvk: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: bl653_dvk: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: nrf52833dk: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: we_ophelia1ev_nrf52805: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: ubx_bmd330eval_nrf52810: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: ubx_bmd360eval_nrf52811: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: pan1782_evb: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: raytac_mdbt50q_db_33_nrf52833: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: bbc_microbit_v2: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: ubx_evkninab4_nrf52833: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: arduino_nano_33_ble: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: adafruit_feather_nrf52840: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: adafruit_itsybitsy_nrf52840: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: bt610: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: bt510: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: mg100: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: pan1780_evb: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: raytac_mdbt50q_db_40_nrf52840: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: pan1770_evb: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: ubx_bmd345eval_nrf52840: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: ubx_evkninab3_nrf52840: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: ubx_bmd340eval_nrf52840: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: ubx_bmd380eval_nrf52840: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: pinnacle_100_dvk: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: particle_boron: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: rak5010_nrf52840: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: particle_xenon: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: rak4631_nrf52840: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: reel_board: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: nrf21540dk_nrf52840: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: particle_argon: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: nrf52840dongle_nrf52840: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: nrf52840_papyr: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: we_proteus3ev_nrf52840: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: nrf52840_mdk_usb_dongle: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: nrf52840_blip: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: nrf52840_mdk: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: pinetime_devkit0: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
cmake: print identifier instead of variant

Update board info printing to use the common term identifier.
Variant is used for the final part of the identifier when a board has
multiple build variants.

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: ubx_evkannab1_nrf52832: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: ubx_bmd300eval_nrf52832: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: nrf52832_mdk: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: ruuvi_ruuvitag: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: thingy52_nrf52832: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: nrf52_adafruit_feather: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: 96b_nitrogen: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: nrf52_vbluno52: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: nrf52_blenano2: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: arduino_nicla_sense_me: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: nrf52_sparkfun: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: ebyte_e73_tbb_nrf52832: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: acn52832: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: we_proteus2ev_nrf52832: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: decawave_dwm1001_dev: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: holyiot_yj16019: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: pan1781_evb: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: ubx_evkninab1_nrf52832: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: blueclover_plt_demo_v2_nrf52832: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: nrf51_vbluno51: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: degu_evk: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: contextualelectronics_abc: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: raytac_mdbt53_db_40_nrf5340: Fix typo

Fixes a typo that ruins grep searches when trying to find
nrf52840-based boards

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: nrf51_ble400: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: bbc_microbit: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: nrf51_blenano: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: rm1xx_dvk: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: nrf51dongle_nrf51422: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: nrf9160dk_*: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: qemu_cortex_m0: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: actinius_icarus_som_dk: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: xiao_ble: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: actinius_icarus_som: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: circuitdojo_feather_nrf9160: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: actinius_icarus: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: actinius_icarus_bee: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: sparkfun_thing_plus_nrf9160: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: nrf9160_innblue22: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: nrf5340_audio_dk_nrf5340: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: nrf9160_innblue21: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: thingy53_nrf5340: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: nrf9131ek_nrf9131: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: bl5340_dvk: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: raytac_mdbt53_db_40: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: raytac_mdbt53v_db_40_nrf5340: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: nrf9151dk_nrf9151: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: nrf9161dk_nrf9161: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
treewide: Update board names for hwmv2

Updates tests that use board names which have changed with boards v2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm commented on boards/v2/96boards/96b_nitrogen/doc/index.rst:
```diff
@@ -0,0 +1,330 @@
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: 96b_nitrogen
```
Some but others no, this one does not

nordicjm committed:
manifest: Update MCUboot

Updates MCUboot to bring in updated hwmv2 board names

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

kartben commented:
Why is the top level folder not "nordic"?
I would think we really want the top-level folder to be something that matches what we have in vendor-prefixes, no? Or to put it differently: what's the expected semantics of `X` in `boards/X` (when X != `shields`, I guess 🙃)

nordicjm committed:
soc: nordic_nrf: Migrate nRF9151 to v2

Moves this SoC over from v1 to v2 so it can be used in HWMv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: 96b_carbon_nrf51: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: nordic_nrf: Migrate nRF54H/nRF54L to v2 and fix nrf

Moves this SoC over from v1 to v2 so it can be used in HWMv2, and
fix the SoC configuration for existing nordic_nrf devices which
was not ported properly

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: nrf54h20pdk_nrf54h20: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: nrf54l15pdk_nrf54l15: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
cmake: modules: configuration_files: Add board identifier overlay file

Fixes an issue in hwmv2 whereby an overlay file exists with the full
board name and identifier which worked in hwmv1

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm commented:
@kartben please re-review

gmarull commented on soc/nordic_nrf/nrf51/Kconfig:
```diff
@@ -7,10 +7,7 @@
 	select HAS_SEGGER_RTT if ZEPHYR_SEGGER_MODULE
 	select HAS_POWEROFF
-	help
-	  Enable support for NRF51 MCU series
```
why does 87c82c111c46bda97607048e8a320b7bdf950897 contain changes to other soc series than 54?

nordicjm replied:
Because the original port for nordic was not done properly and this was needed to actually port it properly

gmarull replied:
then please, split commits, amend original commits, change message or anything else that makes things clear.

nordicjm replied:
Commit message updated

gmarull commented on soc/nordic_nrf/nrf54h/CMakeLists.txt:
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2024 Nordic Semiconductor
+# SPDX-License-Identifier: Apache-2.0
+
+if(CONFIG_ARM)
+  add_subdirectory(arm)
+elseif(CONFIG_RISCV)
+  add_subdirectory(riscv)
+endif()
```
why is this needed?

nordicjm replied:
to include the cmake file for the correct core

gmarull commented on soc/nordic_nrf/nrf54h/arm/Kconfig.soc:
```diff
@@ -0,0 +1,35 @@
+# Copyright (c) 2024 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_SERIES_NRF54HX
+	bool
+	select SOC_FAMILY_NRF
+	help
+	  Nordic Semiconductor nRF54H series MCU
```
if we preserve the arch folder split under soc/nordic_nrf, we'll need to keep duplicating stuff like that, which to me, seems unnecessary. IMHO, we should have all CPUs defined in a single `Kconfig.soc`, regardless of their arch.

nordicjm replied:
Have moved the Kconfig files to the nrf54h directory, the directory split is still needed for arm and risc-specific files e.g. soc.c

gmarull replied:
why don't we simply have `soc_arm.c` and `soc_riscv.c`? As said, the directory split and implied Kconfig duplication just feels wrong. Moving to a hwmv2 where there are arch folders within vendor soc folder isn't a great model.

nordicjm replied:
What file would `#include <soc.h>` which is used all over the place include?

gmarull replied:
Nordic RISC-V code doesn't have `soc.h` since it's fortunately exempt from the [CMSIS mess](https://github.com/zephyrproject-rtos/zephyr/issues/57246). In fact, nobody should be using `soc.h` now that there's `cmsis_core.h` including it. The whole soc.h idea is so brain-damaging.

gmarull replied:
they should simply have a single soc.h, it's now ignored by RISC-V arch code see e7cc2fafb49c1b760bdd85ebf2bc67c53b30c031. If they have unnecessary stuff in soc.h, they should just put it somewhere else. soc.h should just exist to include the "CMSIS glue code" (the entire HAL in most cases)

gmarull replied:
Btw, the whole soc.h scheme is so messed up that stuff like https://github.com/zephyrproject-rtos/zephyr/commit/00566d64fa901cbd7725068f6ff82ee0f0fffc16 is needed because soc.h pollutes almost every single compilation unit.

nordicjm replied:
then you would still have all these files: https://github.com/tejlmand/zephyr/tree/nrf52_adopt_soc_scheme_v2/soc/nordic_nrf/nrf54h/riscv/common/vpr

Renaming them all to `riscv_<filename>` would look like a mess

gmarull replied:
These files should be under `soc/nordic_nrf/common/vpr`, since the VPR core is present in eg nRF54L15 (not added yet). In the future, we should think about supporting RISC-V core setups under `arch/riscv/cores` or something similar, we could have scenarios where a certain RISC-V core implementation is supported by multiple different vendor SoCs.

gmarull commented on soc/nordic_nrf/nrf54h/riscv/Kconfig.soc:
```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2024 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_SERIES_NRF54HX
+	bool
+	select SOC_FAMILY_NRF
+	help
+	  Nordic Semiconductor nRF54H series MCU
```
see my comment above

gmarull commented on soc/nordic_nrf/nrf54h/arm/Kconfig:
```diff
@@ -3,32 +3,29 @@
-
-choice
-	prompt "nRF54H20 MCU Selection"
+config SOC_SERIES_NRF54HX
+	select ARM
+	select ARMV8_M_DSP
+	select CPU_CORTEX_M33
+	select HAS_NRFX
+	select HAS_NORDIC_DRIVERS
+	select HAS_SEGGER_RTT if ZEPHYR_SEGGER_MODULE
```
I find this confusing, because the NRF54HX SoC series no longer can select ARM (or CPU_CORTEX_M33) in a multi-arch setup. These options should be selected by each CPU, even if it implies duplication (imagine eg we had a CORTEX-M33, a CORTEX-M0 and a VPR RISCV).

nordicjm replied:
Split it up properly (this would be easy to spot if there was a board supporting cpuppr)

nordicjm committed:
boards: posix: nrf_bsim: Update paths

Updates paths for the moved nordic_nrf SoC folder

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

gmarull reviewed (approved):
ack for nrf54x changes, a post-hwmv2 cleanup can likely improve soc/nordic_nrf a bit

tejlmand commented on cmake/modules/configuration_files.cmake:
```diff
@@ -77,8 +77,20 @@ zephyr_file(CONF_FILES ${APPLICATION_CONFIG_DIR}/boards DTS APP_BOARD_DTS SUFFIX
 
 zephyr_get(DTC_OVERLAY_FILE SYSBUILD LOCAL)
 if(NOT DEFINED DTC_OVERLAY_FILE)
+  set(board_with_identifier_overlay)
+
+  if(DEFINED BOARD_IDENTIFIER)
+    # Generate overlay filename of board with identifier present
+    zephyr_build_string(board_with_identifier_overlay
+                        BOARD ${BOARD}
+                        BOARD_IDENTIFIER ${BOARD_IDENTIFIER}
+    )
+
+    set(board_with_identifier_overlay "${board_with_identifier_overlay}.overlay")
+  endif()
```
How about:
```suggestion
  zephyr_build_string(board_overlay_strings
                      BOARD ${BOARD}
                      BOARD_IDENTIFIER ${BOARD_IDENTIFIER}
                      MERGE
  )
  list(TRANSFORM board_overlay_strings APPEND ".overlay")
```
and then use `board_overlay_strings` below in:

```
        NAMES "${APP_BOARD_DTS};${board_overlay_strings };app.overlay" SUFFIX ${FILE_SUFFIX})
```

nordicjm replied:
Fixed

nordicjm committed:
boards: arm: pan1783: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>
